engine/src/test/java/org/hibernate/validator/test/internal/engine/ValidatorTest.java

```diff
@@ -325,6 +344,20 @@ public String getFoo() {
 		}
 	}
 
+	class I {
+		@Valid
+		Map<String, J> map = new HashMap<String, J>();
```

Comment:
CollectionHelper was created at the time where the diamond didn't exist so it was a convenient method to initialize a generic collection. It also has helpers to create sized collection. It has a little less sense now that we have the diamond but we still use it as the primary way to initialize collection.

In this case, I must admit that I like to keep it simple for this sort of test but it's a matter of preference. And I didn't use the diamond operator as I plan to backport this on 5.3/5.4.